python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -218,10 +288,36 @@ def _init_nccl_group(
 
     logger.info("NCCL group created.")
 
-    ctx.nccl_group = _NcclGroup(
+    ctx.nccl_groups[group_id] = _NcclGroup(
         world_size,
-        comm_id,
+        nccl_comm_id,
         rank=None,
-        actor_ids_to_ranks=actor_ids_to_ranks,
+        actor_handles=actors,
         cuda_stream=None,
     )
+    return group_id
+
+
+def _destroy_nccl_group(group_id: str) -> None:
+    ctx = ChannelContext.get_current()
+    if group_id not in ctx.nccl_groups:
+        return
+
+    group = ctx.nccl_groups[group_id]
+    actors = group._get_actor_handles()
+    destroy_tasks = [
+        actor.__ray_call__.remote(
+            _do_destroy_nccl_group,
+            group_id,
+        )
+        for actor in actors
+    ]
+
+    _, unready = ray.wait(destroy_tasks, timeout=30, num_returns=len(destroy_tasks))
+    if unready:
+        logger.warning(
```

Comment:
Should we kill actors this case? Maybe actors are just leaked otherwise?

Reply:
We want to be able to reuse the actors.

Reply:
My concern here is if this doesn't return on time, doesn't this mean there's sth wrong with nccl, and reusing actors potentially can cause some weird issues?